### PR TITLE
Fix creation and deletion of vdis

### PR DESCRIPTION
**Pull request by MatiasVara**
I open this PR to discuss changes to support new behavior on fsp vol driver for the creation and deletion of volumes (see https://github.com/xcp-ng/xcp-ng-xapi-storage/pull/9).

### CODE REVIEW

File: files/delete
Comment (MatiasVara) [inline files/delete]: For FSP, vdi-destroy has been removed. This needs to be fixed.

File: files/delete
Comment (MatiasVara) [inline files/delete]: After removing the symlink, we require to issue a scan() to do the actual vdi_destroy.

File: files/delete
Comment (MatiasVara) [inline files/delete]: The name of the directory has changed. Fix it.

File: files/create
Comment (MatiasVara) [inline files/create]: The name of the directory has changed. Fix it.

**Comment by Wescoeur** [comment]
Note: we must fix remaining issues here before merging https://github.com/xcp-ng/xcp-ng-tests/pull/97 and https://github.com/xcp-ng/xcp-ng-xapi-storage/pull/9 .

File: files/create
Comment (Wescoeur) [inline files/create]: Use the VM UUID instead, it's more robust. ;)
```bash
ln -s "$mountpoint" "$dir_links/$vm_uuid"
```

File: files/delete
Comment (Wescoeur) [inline files/delete]:
```bash
rm -f "$dir_links/$vm_uuid"
```

File: files/create
Comment (Wescoeur) [inline files/create]: `a scan`

File: files/delete
Comment (Wescoeur) [inline files/delete]: You can remove this code if you store the SR path in a file like for VM and VDI uuids. ;)
Search in `files/create`:
```bash
echo -n "$vm_uuid" > "$vm_uuid_file"
echo -n "$vdi_uuid" > "$vdi_uuid_file"
```

**Review by Wescoeur — CHANGES_REQUESTED** [review]

**Review by benjamreis — APPROVED** [review]
LGTM after @Wescoeur's required modifications. :)

File: files/delete
Comment (MatiasVara) [reply files/delete]: I see. I am a bit confused if I should use a file or a parameter to pass the sr path. For example, in `files/create`, the script is called with the following parameters:
```bash
"$workpath/container" "$workpath" "$vm_uuid" "$vdi_uuid" "$wrapper_pid" "$container_pipe" &
```
Then, in the `container` script, you have:
```bash
"$workpath/delete" "$vm_uuid" "$vdi_uuid"
```
And finally, in the `delete` script, there is:
```bash
vm_uuid=$1
vdi_uuid=$2
```
I am probably missing something but what mechanism should I use? Maybe, it is better to use arguments?

Comment (Wescoeur) [reply files/delete]: The goal is to store the SR path in a file, so you can modify this code in `runx` file:
```python
    if [ -f "$crundir/xcp_vm_uuid" ] && [ -f "$crundir/xcp_vdi_uuid" ]; then
        "$workpath/delete" `cat "$crundir"/xcp_vm_uuid` `cat "$crundir"/xcp_vdi_uuid`
        "$workpath/mount" "$crundir" unmount
    fi
```

You can call the delete script with:
```python
"$workpath/delete" "$vm_uuid" "$vdi_uuid" "$sr_path"
```

So to resume:
  - We can store the path in a file during VDI creation
  - It's probably better to call the delete script with a path argument, we can read it like other params in the main script

File: runx
Comment (Wescoeur) [inline runx]: I wonder if we must keep this check with a test on the new xcp_sr_path variable or if it's a good idea to remove it totally. @benjamreis

File: files/delete
Comment (Wescoeur) [inline files/delete]: It's not required to ask to xe for the VDI UUID. You can save this attribute in a file like the path.
Note: if we have to add other information in the future, it would probably be good to use a json file for example. But not in this PR. :wink:

File: files/create
Comment (Wescoeur) [inline files/create:101]: Ok for this PR, not a big issue. We can fix this piece later.

File: files/delete
Comment (Wescoeur) [reply files/delete]: What do you think @benjamreis? I don't think it's a good idea to break symmetry with the path. Saving both information seems to me the correct way.

**Review by Wescoeur — CHANGES_REQUESTED** [review]

File: runx
Comment (MatiasVara) [reply runx]: I was also wondering that.

File: files/delete
Comment (MatiasVara) [reply files/delete]: I saved the sr_uuid in a file.

File: runx
Comment (MatiasVara) [reply runx]: Those files are created by the `files/create` script. If that script fails or the `runx` script fails before invoking `files/create`, those files are not created. In any case, `runx delete` is invoked and the check prevents the execution of `files/delete`. These files are created at the same time in `files/create` so checking the existence of one would be enough to ensure that the other exists if there is no issue during the creation itself, e.g., running out of disk space. Also, the check only checks the existence of the files but not if the content is valid. Based on this, I think we could simply check also the existence of the `xcp_sr_path` file and the `xcp_sr_uuid` file.

**Review by benjamreis — APPROVED** [review]

**Review by Wescoeur — APPROVED** [review]

**Pull request merged**
